superset/databases/api.py

```diff
@@ -1433,51 +1433,15 @@ def oauth2(self) -> FlaskResponse:
               $ref: '#/components/responses/500'
         """
         parameters = OAuth2ProviderResponseSchema().load(request.args)
+        command = OAuth2StoreTokenCommand(parameters)
+        command.run()
 
-        if "error" in parameters:
-            raise OAuth2Error(parameters["error"])
-
-        # note that when decoding the state we will perform JWT validation, preventing a
-        # malicious payload that would insert a bogus database token, or delete an
-        # existing one.
         state = decode_oauth2_state(parameters["state"])
+        tab_id = state["tab_id"]
```

Comment:
I see we're currently calling `decode_oauth2_state` both here at the API layer and also at the command. Should the `tab_id` be returned by the command directly? Not super opinionated here, just wondering your thoughts.

Reply:
Yeah, I agree this is not ideal. But I made the command return the created model instance, and unfortunately it doesn't have `tab_id` associated with it. I thought it was better to keep the interface simple, and have a bit of duplicity here.

Reply:
that make sense. thank you!